pkg/goDB/conditions/node/filter_direction_test.go

```diff
@@ -0,0 +1,132 @@
+		expectedFilter:    types.Counters.IsBidirectional, expectedErr: nil},
+	// invalid filter (2 traffic filters provided)
+	{node: andNode{left: conditionNode{attribute: "dir", comparator: "=", value: "uni"},
+		right: conditionNode{attribute: "dir", comparator: "=", value: "bi"}},
+		expectedCondition: nil,
+		expectedFilter:    nil, expectedErr: errMultipleDirectionFilterConditions},
+	// invalid filter (2 traffic filters provided, of which one is nested)
+	{node: andNode{left: conditionNode{attribute: "dir", comparator: "=", value: "uni"},
+		right: andNode{left: conditionNode{attribute: "dip", comparator: "=", value: "127.0.0.1"}, right: conditionNode{attribute: "dir", comparator: "=", value: "bi"}}},
+		expectedCondition: nil,
+		expectedFilter:    nil, expectedErr: errMultipleDirectionFilterConditions},
+	// simple condition without any traffic filter => valid
+	{node: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"},
+		expectedCondition: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"},
+		expectedFilter:    nil, expectedErr: nil},
+	// disjunctive condition with traffic filter => invalid
+	{node: orNode{left: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"}, right: conditionNode{attribute: "dir", comparator: "=", value: "in"}},
+		expectedCondition: nil,
+		expectedFilter:    nil, expectedErr: errMisplacedDirectionFilterCondition},
+	// conjunctive condition without any traffic filter => valid
+	{node: andNode{left: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"}, right: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"}},
+		expectedCondition: andNode{left: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"}, right: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"}},
+		expectedFilter:    nil, expectedErr: nil},
+	// conjunctive condition with misplaced traffic filter => invalid
+	{node: andNode{left: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"},
+		right: orNode{left: conditionNode{attribute: "sip", comparator: "=", value: "127.0.0.1"}, right: conditionNode{attribute: "dir", comparator: "=", value: "in"}}},
+		expectedCondition: nil,
+		expectedFilter:    nil, expectedErr: errMisplacedDirectionFilterCondition},
+}
+
+func TestSplitOffDirectionFilter(t *testing.T) {
+	for _, test := range splitOffDirectionFilterTest {
+		condition, valFilterNode, err := splitOffDirectionFilter(test.node)
+		if err != nil && err.Error() != test.expectedErr.Error() {
+			t.Fatalf("Expected error: %s Actual error: %s\n", test.expectedErr, err)
+		}
+
+		if err == nil && reflect.ValueOf(valFilterNode.ValFilter) != reflect.ValueOf(test.expectedFilter) {
+			t.Fatalf("Expected filter type: %s Got: %s", reflect.TypeOf(test.expectedFilter), reflect.TypeOf(valFilterNode.ValFilter))
+		}
+		if err == nil && condition.String() != test.expectedCondition.String() {
+			t.Log("Incorrect split off of condition")
```

Comment:
These four lines can be condensed into a single one using `testify`'s built-in formatting directives.